lua/modes.lua

```diff
@@ -28,91 +54,67 @@ end
 ---Update highlights
 ---@param scene 'default'|'insert'|'visual'|'copy'|'delete'|
 M.highlight = function(scene)
-	if scene == 'default' then
-		utils.set_hl('CursorLine', { bg = default_colors.cursor_line })
-		if config.set_number then
-			utils.set_hl('CursorLineNr', { bg = default_colors.cursor_line_nr })
-		end
-		utils.set_hl('ModeMsg', { fg = default_colors.mode_msg })
-		utils.set_hl('Visual', { bg = default_colors.visual })
-	end
-
-	if scene == 'insert' then
-		utils.set_hl('CursorLine', { bg = blended_colors.insert })
-		if config.set_number then
-			utils.set_hl('CursorLineNr', { bg = blended_colors.insert })
+	local winhl_map = {}
+	local old_value = vim.api.nvim_win_get_option(0, 'winhighlight')
```

Comment:
I would suggest "prev" over "old" if this value is continually updating to be the last value. Otherwise "old" is fine

```suggestion
	local prev_value = vim.api.nvim_win_get_option(0, 'winhighlight')
```